addons/test/misra/misra-test.c

```diff
@@ -233,6 +233,26 @@ void misra_7_3() {
   long double misra_7_3_e = 7.3l; //7.3
   }
 
+typedef const char* MISRA_7_4_CHAR_CONST;
+MISRA_7_4_CHAR_CONST misra_7_4_return_const (void) { return "return_typedef_const"; } // 18.4
+char *misra_7_4_return_non_const (void) { return 1 + "return_non_const"; } // 7.4 18.4
+const char *misra_7_4_return_const (void) { return 1 + "return_const"; } // 18.4
+
+void misra_7_4_const_call(int a, const char* b) { }
+void misra_7_4_call(int a, char* b) { }
+
+void misra_7_4()
```

Comment:
From what I could tell, 7.4 also applies to wide string (`wchar_t`). Perhaps add some test cases for that as well?

Reply:
Thanks for the input. I have added a test case for this in my latest commit.